components/pages/submission-system/program-dashboard/DonorDataSummary/index.tsx

```diff
@@ -38,27 +39,45 @@ import DownloadButtons from './DownloadButtons';
 import { InvalidDonorsNotification } from './InvalidDonorsNotification';
 import ContentError from '../../../../placeholders/ContentError';
 
-export const useProgramDonorsSummaryQuery = (
-  programShortName: string,
-  first: number,
-  offset: number,
-  sorts: DonorSummaryEntrySort[],
-  options: Omit<
+export const useProgramDonorsSummaryQuery = ({
+  programShortName,
+  first,
+  offset,
+  sorts,
+  filters,
+  options = {},
+}: {
+  programShortName: string;
+  first: number;
+  offset: number;
+  sorts?: DonorSummaryEntrySort[];
+  filters?: ProgramDonorSummaryFilter[];
+  options?: Omit<
     QueryHookOptions<ProgramDonorsSummaryQueryData, ProgramDonorsSummaryQueryVariables>,
     'variables'
-  > = {},
-) => {
+  >;
+}) => {
   const pollingTimeout = useTimeout(30000);
+  const variables =
+    filters && Object.keys(filters).length > 0
+      ? {
+          programShortName,
+          first,
+          offset,
+          sorts,
+          filters,
+        }
+      : {
+          programShortName,
+          first,
+          offset,
+          sorts,
+        };
```

Comment:
This can also be expressed as:

```suggestion
  const variables = {
    programShortName,
    first,
    offset,
    sorts,
    ...(filters && Object.keys(filters).length > 0 && { filters }),
  };
```

which avoids repetition

Reply:
🤯 this is so much cleaner!